### PR TITLE
[PL-48049] DEL 82502 Release Notes

### PR DESCRIPTION
**Pull request by saifeemustafaqharness**
[PL-48049](https://harness.atlassian.net/browse/PL-48049)

https://harness.atlassian.net/wiki/spaces/PD/pages/21684125729/NG+Release+SAAS+March+4th+2024+for+1.28.x-+Build#NG-Delegate-dependent-changes--

For reviewers, a preview is available:

https://65f1c95bfbe0d13aca97aafa--harness-developer.netlify.app/release-notes/delegate#harness-version-12811-harness-delegate-version-240382502-

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screenshot. 

[PL-48049]: https://harness.atlassian.net/browse/PL-48049?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by saifeemustafaqharness** [comment]
@pratmit @brian-f-harness Ready for review

File: release-notes/delegate.md [hunk 1]
```diff
@@ -71,6 +71,32 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
 
 ## March 2024
 
+### Harness version 1.28.9, Harness Delegate version 24.02.82501 <!--  February 27, 2024 -->
```
Comment (brian-f-harness) [inline release-notes/delegate.md:74]:
```suggestion
### Harness version 1.28.9, Harness Delegate version 24.02.82501 <!--  March xx, 2024 -->
```

File: release-notes/delegate.md [hunk 2]
```diff
@@ -71,6 +71,32 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+
+#### New features and enhancements
+
+- Introduced separate environment variables to manage delegate resource thresholds for CPU and Memory when dynamic handling is enabled. (PL-47746)
+
+   - Use CPU_USAGE_THRESHOLD for CPU control (default: no limit)
```
Comment (brian-f-harness) [inline release-notes/delegate.md:80]:
```suggestion
   - Use `CPU_USAGE_THRESHOLD` for CPU control (default: no limit)
```

File: release-notes/delegate.md [hunk 3]
```diff
@@ -71,6 +71,32 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+
+   - Use MEMORY_USAGE_THRESHOLD for memory control (default: 80%)
```
Comment (brian-f-harness) [inline release-notes/delegate.md:82]:
```suggestion
   - Use `MEMORY_USAGE_THRESHOLD` for memory control (default: 80%)
```

File: release-notes/delegate.md [hunk 4]
```diff
@@ -71,6 +71,32 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+
+   If you are using RESOURCE_USAGE_THRESHOLD (deprecated), it exclusively controls the memory threshold.
```
Comment (brian-f-harness) [inline release-notes/delegate.md:84]:
```suggestion
   If you are using `RESOURCE_USAGE_THRESHOLD` (deprecated), it exclusively controls the memory threshold.
```

File: release-notes/delegate.md [hunk 5]
```diff
@@ -71,6 +71,32 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+
+- OPA policy enforcement has been introduced to three new entities, namely Service Accounts, API Keys, and Tokens. For Service Accounts and API Keys, naming conventions policies are enforced, while for Tokens, Time-To-Live (TTL) policies are enforced. These enforcement mechanisms are seamlessly integrated into both create and update operations, ensuring adherence to predefined standards during the onSave action. (PL-46778)
```
Comment (brian-f-harness) [inline release-notes/delegate.md:86]:
```suggestion
- OPA policy enforcement has been introduced to three new entities: Service Accounts, API Keys, and Tokens. For Service Accounts and API Keys, naming convention policies are enforced, while for Tokens, Time-To-Live (TTL) policies are enforced. These enforcement mechanisms are seamlessly integrated into both create and update operations, ensuring adherence to predefined standards during the `onSave` action. (PL-46778)
```

File: release-notes/delegate.md [hunk 6]
```diff
@@ -71,6 +71,32 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+
+- Support added to enable OPA policy for naming convention enforcement while creating or updating a Service Account. (PL-46777)
```
Comment (brian-f-harness) [inline release-notes/delegate.md:88]:
```suggestion
- Support added to enable OPA policy for naming convention enforcement while creating or updating a service account. (PL-46777)
```

File: release-notes/delegate.md [hunk 7]
```diff
@@ -71,6 +71,32 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+
+#### Fixed issues
+
+- Attempts to use the `harness_platform_user` resource to create or delete users results in an error. The message "Request failed as you have an older version of entity, please reload the page and try again" is displayed and the Terraform state goes out of sync with Harness. (PL-39870, ZD-47107)
```
Comment (brian-f-harness) [inline release-notes/delegate.md:92]:
```suggestion
- Attempts to use the `harness_platform_user` resource to create or delete users resulted in an error. The message "Request failed as you have an older version of an entity, please reload the page and try again" was displayed and the Terraform state went out of sync with Harness. (PL-39870, ZD-47107)
```

File: release-notes/delegate.md [hunk 8]
```diff
@@ -71,6 +71,32 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
 
 ## March 2024
 
+### Harness version 1.28.9, Harness Delegate version 24.02.82501 <!--  February 27, 2024 -->
+
```
Comment (brian-f-harness) [inline release-notes/delegate.md:75]: Please verify the delegate verno. I see 24.03.82502 in QA.

**Review by brian-f-harness — CHANGES_REQUESTED** [review]

**Review by brian-f-harness — APPROVED** [review]

File: release-notes/delegate.md [hunk 9]
```diff
@@ -2,7 +2,7 @@
 title: Delegate release notes
 sidebar_label: Delegate
 tags: [NextGen, "Delegate"]
-date: 2024-03-08T10:00
+date: 2024-03-12T10:00
```
Comment (brian-f-harness) [inline release-notes/delegate.md:5]:
```suggestion
date: 2024-03-13T10:00
```

File: release-notes/delegate.md [hunk 10]
```diff
@@ -71,6 +71,32 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
 
 ## March 2024
 
+### Harness version 1.28.10, Harness Delegate version 24.03.82502 <!--  March 12, 2024 -->
```
Comment (brian-f-harness) [inline release-notes/delegate.md:74]:
```suggestion
### Harness version 1.28.10, Harness Delegate version 24.03.82502 <!--  March 13, 2024 -->
```

**Review by brian-f-harness — APPROVED** [review]
LGTM

File: release-notes/delegate.md [hunk 11]
```diff
@@ -71,6 +71,32 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
 
 ## March 2024
 
+### Harness version 1.28.10, Harness Delegate version 24.03.82502 <!--  March 13, 2024 -->
```
Comment (brian-f-harness) [inline release-notes/delegate.md:74]:
```suggestion
### Harness version 1.28.11, Harness Delegate version 24.03.82502 <!--  March 13, 2024 -->
```

**Review by brian-f-harness — CHANGES_REQUESTED** [review]
The deployed version is 1.28.11

https://harness.slack.com/archives/C04R4UDE1FT/p1710307241653819?thread_ts=1710211576.561939&cid=C04R4UDE1FT

**Review by brian-f-harness — APPROVED** [review]

**Review by pratmit — APPROVED** [review]
LGTM

**Pull request merged**